src/Framework/BuildCheck/BuildCheckEventArgs.cs

```diff
@@ -107,7 +107,8 @@ internal override void CreateFromStream(BinaryReader reader, int version)
 }
 internal sealed class BuildCheckResultWarning : BuildWarningEventArgs
 {
-    public BuildCheckResultWarning(IBuildCheckResult result)
+    public BuildCheckResultWarning(IBuildCheckResult result, string code)
+        : base(subcategory: null, code: code, file: null, lineNumber: 0, columnNumber: 0, endLineNumber: 0, endColumnNumber: 0, message: result.FormatMessage(), helpKeyword: null, senderName: null)
```

Comment:
In the future we should populate the location here from the result.Location (that currently is not structured, so bit of refactor will be needed).

Not needed in a scope of this PR